### PR TITLE
Upgrade: Directory to Ember v3.7

### LINKED ISSUE #192: Update Directory to the latest ember version
Update Directory to the latest ember version

### PR DESCRIPTION
**Pull request by daniellevinson**
Resolves #192

## Description
Upgrade navi-directory to Ember 3.7

### CODE REVIEW

File: packages/directory/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 'use strict';
 
 module.exports = {
-  name: 'navi-directory'
+  name: require('./package').name
```
Comment (cythrawll) [inline packages/directory/index.js]: what's going on here, did a codemod do this?  Does this really resolve to the same thing?

Comment (cythrawll) [reply packages/directory/index.js]: nm I see that it would end up being the same thing.

File: packages/directory/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - env: EMBER_TRY_SCENARIO=ember-canary
+    - env: EMBER_TRY_SCENARIO=ember-default-with-jquery
+
+before_install:
+  - npm config set spin false
+  - npm install -g npm@4
+  - npm --version
+
+script:
+  - node_modules/.bin/ember try:one $EMBER_TRY_SCENARIO
```
Comment (Alex-Aralis) [inline packages/directory/.travis.yml]: This file is not used.

File: packages/directory/addon/components/dir-asset-row-actions.js [hunk 3]
```diff
@@ -24,9 +24,9 @@ export default Component.extend({
    * @property {String} type - the type of the asset
    */
   type: computed('value', function() {
-    let value = get(this, 'value');
+    let { value } = this;
```
Comment (Alex-Aralis) [inline packages/directory/addon/components/dir-asset-row-actions.js]: In the past destructures have failed for me when the 3.x addons have been used with 2.x apps.  Does this work with zig or teller?

File: packages/directory/addon/mixins/search-filter.js [hunk 4]
```diff
@@ -19,8 +19,6 @@ export default Mixin.create({
    */
   searchResults: computed('directory.q', 'sortedItems', function() {
     let queryString = get(this, 'directory.q');
-    return get(this, 'sortedItems').then(
-      items => (isEmpty(queryString) ? items : searchRecords(items, queryString, 'title'))
-    );
+    return this.sortedItems.then(items => (isEmpty(queryString) ? items : searchRecords(items, queryString, 'title')));
```
Comment (Alex-Aralis) [inline packages/directory/addon/mixins/search-filter.js]: `(isEmpty(queryString))` => `isEmpty(queryString)`

File: packages/directory/config/ember-try.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+          npm: {
+            devDependencies: {
+              '@ember/jquery': '^0.5.1'
+            }
+          }
+        }
+      ]
+    };
+  });
+};
```
Comment (Alex-Aralis) [inline packages/directory/config/ember-try.js]: File not used.

File: packages/directory/package.json [hunk 6]
```diff
@@ -6,87 +6,95 @@
...
   "devDependencies": {
-    "broccoli-asset-rev": "^2.4.5",
-    "ember-ajax": "^3.0.0",
+    "@ember/jquery": "^0.6.0",
+    "@ember/optional-features": "^0.6.3",
+    "broccoli-asset-rev": "^2.7.0",
     "ember-basic-dropdown": "1.0.1",
     "ember-c3": "0.2.3",
-    "ember-cli": "^3.1.4",
+    "ember-cli": "~3.7.1",
```
Comment (Alex-Aralis) [inline packages/directory/package.json]: can we go up to 3.8?  nothing much changes.

File: packages/directory/addon/components/dir-asset-row-actions.js (hunk 3 above)
Comment (daniellevinson) [reply packages/directory/addon/components/dir-asset-row-actions.js]: teller is already upgraded but I'll check with zig

File: packages/directory/config/ember-try.js (hunk 5 above)
Comment (daniellevinson) [reply packages/directory/config/ember-try.js]: do we need the `ember-try` dependency then? a standard addon has a `"test:all": "ember try:each"` npm script

File: packages/directory/addon/components/dir-asset-row-actions.js (hunk 3 above)
Comment (mattgraham1995) [reply packages/directory/addon/components/dir-asset-row-actions.js]: Ziggurat doesn't use directory anyway

Comment (daniellevinson) [reply packages/directory/addon/components/dir-asset-row-actions.js]: I've probably changed to this syntax in a few other places

File: packages/directory/config/ember-try.js (hunk 5 above)
Comment (Alex-Aralis) [reply packages/directory/config/ember-try.js]: hmm, not sure.  The tests are being run by lerna via the top level package.json.  I don't think we are currently using ember-try in our tests.

Comment (cythrawll) [reply packages/directory/config/ember-try.js]: I kind of think we should keep these things in place even if we don't use them.  Ember upgrades will just put them back, and nott worth the effort to remember to remove them. let's reduce what needs to be done with each upgrade.

File: packages/directory/addon/components/dir-asset-row-actions.js (hunk 3 above)
Comment (daniellevinson) [reply packages/directory/addon/components/dir-asset-row-actions.js]: oh wait I forgot it's only a directory upgrade :laughing:

File: packages/directory/config/ember-try.js (hunk 5 above)
Comment (Alex-Aralis) [reply packages/directory/config/ember-try.js]: I tried it with core and it works fine.  Reverting for now.

File: packages/directory/addon/components/dir-asset-row-actions.js (hunk 3 above)
Comment (daniellevinson) [reply packages/directory/addon/components/dir-asset-row-actions.js]: Teller's looking good so far

File: packages/directory/addon/mixins/search-filter.js (hunk 4 above)
Comment (daniellevinson) [reply packages/directory/addon/mixins/search-filter.js]: It's a `prettier` thing

File: packages/directory/config/ember-try.js (hunk 5 above)
Comment (daniellevinson) [reply packages/directory/config/ember-try.js]: I agree with @cythrawll that it makes sense to keep these

File: packages/directory/tests/helpers/start-app.js [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
 
 export default function startApp(attrs) {
-  let attributes = merge({}, config.APP);
+  let attributes = assign({}, config.APP);
   attributes.autoboot = true;
-  attributes = merge(attributes, attrs); // use defaults, but you can override;
+  attributes = assign(attributes, attrs); // use defaults, but you can override;
 
   return run(() => {
     let application = Application.create(attributes);
```
Comment (balajimanoharan) [inline packages/directory/tests/helpers/start-app.js]: Is this file used? This used to be the old way to run acceptance tests.

File: packages/directory/.travis.yml (hunk 2 above)
Comment (jkusa) [reply packages/directory/.travis.yml]: Can we add this to the gitignore?

File: packages/directory/config/ember-try.js (hunk 5 above)
Comment (jkusa) [reply packages/directory/config/ember-try.js]: In general, if we don't want a generated file we should add an entry for it in the gitignore file. 

In this case, this file can be helpful for future upgrades in determining supported Ember versions.

File: packages/directory/tests/helpers/start-app.js (hunk 7 above)
Comment (daniellevinson) [reply packages/directory/tests/helpers/start-app.js]: hmm don't think so, tests still pass when removing this one and destroy-app.js

**Pull request merged**